src/main/java/com/commercetools/sync/commons/utils/GenericUpdateActionUtils.java

```diff
@@ -15,94 +12,59 @@
-     * be a {@link Category} or a {@link Channel}).
+     * Creates a CTP "setCustomType" update action on the given resource {@code T} according to the type of the
+     * {@link GenericCustomActionBuilder} passed. If the {@code customTypeId} passed is blank (null/empty), the error
+     * callback is triggered with an error message that the setCustomType update action cannot be built with a blank id,
+     * and an empty optional is returned.
      *
-     * @param customTypeId       the id of the new custom type.
-     * @param customFieldsJsonMap the custom fields map of JSON values.
-     * @param resource            the resource to do the update action on.
-     * @param <T>                 the type of the resource to do the update action on.
-     * @param syncOptions         responsible for supplying the sync options to the sync utility method.
-     * @return a setCustomType update action of the type of the resource it's requested on.
+     * @param <T>                    the type of the resource which has the custom fields.
+     * @param <U>                    the type of the resource to do the update action on.
+     * @param customTypeId           the id of the new custom type.
+     * @param customFieldsJsonMap    the custom fields map of JSON values.
+     * @param resource               the resource which has the custom fields.
+     * @param customActionBuilder    the builder instance responsible for building the custom update actions.
+     * @param variantId              optional field representing the variant id in case the oldResource is an asset.
+     * @param resourceIdGetter       a function used to get the id of the resource being updated.
+     * @param resourceTypeIdGetter   a function used to get the Type id of the resource being updated.
+     * @param updateIdGetter         a function used to get the id/key needed for updating the resource that has the
+     *                               custom fields.
+     * @param syncOptions            responsible for supplying the sync options to the sync utility method.
+     * @return a setCustomType update action of the type of the resource it's requested on, or an empty optional
+     *         if the {@code customTypeId} is blank.
      */
     @Nonnull
-    static <T extends Custom & Resource<T>> Optional<UpdateAction<T>> buildTypedSetCustomTypeUpdateAction(
+    static <T extends Custom, U extends Resource<U>> Optional<UpdateAction<U>> buildTypedSetCustomTypeUpdateAction(
         @Nullable final String customTypeId,
         @Nullable final Map<String, JsonNode> customFieldsJsonMap,
         @Nonnull final T resource,
+        @Nonnull final GenericCustomActionBuilder<U> customActionBuilder,
+        @Nullable final Integer variantId,
+        @Nonnull final Function<T, String> resourceIdGetter,
+        @Nonnull final Function<T, String> resourceTypeIdGetter,
+        @Nonnull final Function<T, String> updateIdGetter,
```

Comment:
The reason why it's called `updateIdGetter` here is to show that it's the id used for updating the resource for example, in case of assets, it's asset key, in case of prices it's price id. This is different than the `resourceIdGetter` which is merely the uuid of the resource, but its not necesseraily used for the "updating" it. What every param is doing is explained in the Javadoc of the method 😉

Reply:
ok